ADAL/src/cache/ADTokenCache.m

```diff
@@ -567,7 +603,7 @@ - (BOOL)addOrUpdateImpl:(ADTokenCacheItem *)item
 
 - (NSString *)description
 {
-    return [NSString stringWithFormat:@"ADTokenCache: %@", _cache];
+    return [NSString stringWithFormat:@"ADTokenCache: %@", [self cacheCopy]];
```

Comment:
It would be better to grab the lock then create a copy here (which you're going to have to grab to create the copy in the first place...)